libnavigation-base/src/main/java/com/mapbox/navigation/base/route/model/RouteStepProgressNavigation.kt

```diff
@@ -0,0 +1,137 @@
+package com.mapbox.navigation.base.route.model
+
+data class RouteStepProgressNavigation internal constructor(
+    var step: LegStepNavigation? = null,
+    var distanceRemaining: Double? = null,
+    var distanceTraveled: Double? = null,
+    var fractionTraveled: Float = 0f,
+    var durationRemaining: Double? = null,
```

Comment:
**Regarding `private val`**
Yeah thanks for pointing them out.

**Regarding `non-null`**
I would leave it nullable otherwise there is no good way to specify the default value. I can make it 0.0 or -10.0. How do we decide and generalize? Also, there are scenarios, when the directions route API call returns null route for a destination I want to visit. 
We have come across these scenarios earlier and tried to resolve them using `lateinit`, but it starts to throw `NullPointerException`.

Reply:
Thinking it through again, it makes sense to have some of these as non null, but I am not sure as to how they would affect other computations.

Reply:
Looks like we are always calculating some values like distance traveled etc. The DTO doesn't make much sense without this data, so we should just default to 0 in those cases.